Purge transfers should check for state

matszczygiel opened the pull request:
Blocked by https://github.com/NordSecurity/libdrop/pull/191

LukasPukenis commented on drop-storage/src/lib.rs:
```diff
@@ -1721,8 +1773,22 @@ mod tests {
             .purge_transfers(&[transfer_id_1.to_string(), transfer_id_2.to_string()])
             .await;
 
+        // Because the transfers havent' reached termial state
```
```suggestion
        // Because the transfers haven't reached the terminal state
```

LukasPukenis reviewed (approved):
These both PRs look good, however, they are tricky in their behavior. Would be nice to find a way to integrate-test them.

Few ideas:
- Emit internal events via API that would be useless to real API users but only for testing purposes(not that nice)
- Return deleted transfers in the JSON API and mark them as deleted. May return them in a separate object so as not to make API users accidentally use them(nicer)

What is your take on these?

matszczygiel commented:
@LukasPukenis in https://github.com/NordSecurity/libdrop/pull/191 I introduced a unit test to check how many transfers we've removed in the `remove_garbage_transfers()`. I think it somewhat checks all the possible cases
